stock_picking_accounting_date/models/stock_picking.py

```diff
@@ -12,12 +12,23 @@ class StockPicking(models.Model):
     )
     show_accounting_date = fields.Boolean(compute="_compute_show_accounting_date")
 
+    def _check_internal_or_transit_location_type(self, locations):
+        self.ensure_one()
+        locations = set(locations)
+        if any(location.usage not in ["internal", "transit"] for location in locations):
+            return False
+        return True
+
     def _compute_show_accounting_date(self):
         for pick in self:
             pick.show_accounting_date = False
-            if pick.picking_type_code not in ("incoming", "outgoing"):
-                continue
-            if pick.move_ids.with_company(pick.company_id).product_id.filtered(
-                lambda x: x.detailed_type == "product" and x.valuation == "real_time"
-            ):
-                pick.show_accounting_date = True
+            move_ids = pick.move_ids.with_company(pick.company_id).filtered(
+                lambda move: move.product_id.detailed_type == "product"
+                and move.product_id.valuation == "real_time"
+            )
+            if move_ids:
+                move_source_locations = [move.location_id for move in move_ids]
+                move_dest_locations = [move.location_dest_id for move in move_ids]
+                locations = move_source_locations + move_dest_locations
+                if not self._check_internal_or_transit_location_type(locations):
+                    pick.show_accounting_date = True
```

Comment:
We need to consider the combination of source and destination locations to judge whether there will be valuation records.

```suggestion
            valued_moves = pick.move_ids.with_company(pick.company_id).filtered(
                lambda move: move.product_id.detailed_type == "product"
                and move.product_id.valuation == "real_time"
                and not (move.location_id._should_be_valued() and move.location_dest_id._should_be_valued())
            )
            if valued_moves:
                pick.show_accounting_date = True
```